### DISCUSSION TITLE
`quarto self-update`

### DISCUSSION THREAD

**Discussion by kdheepak** [quarto-dev/quarto-cli#11026] [Feature Requests]
### Description

It'd be nice to have a subcommand to `quarto` that updates the quarto installation. Both `pixi` and `uv` have this, and I find it extremely convenient to update them. With quarto, I have to follow a different process on macos and windows.

**Comment by cscheid** [comment]
Agreed!

But exactly because there's different processes on macos and windows, it's a big side quest for us to figure out how to make the feature work across the different ways that Quarto is packaged. So we don't plan to do it on at least the next 12-18 months.

**Reply by kdheepak** [reply]
Makes sense, I imagine it is non-trivial. 

As a follow up to the command line option, I can also imagine it be useful to make it easier to update the install from RStudio / VSCode etc.

Would you be able to elaborate on the different ways Quarto is packaged? I see there are `.tar.gz` files (but haven't tried them).

**Reply by cscheid** [reply]
If you don't mind me being frank - I don't have the time to have that discussion in full. But:

<img width="495" alt="image" src="https://github.com/user-attachments/assets/bcdc0c16-f8bf-4a8c-8a15-f74dc925aed2">

And then third-party packaging. Off the top of my head we have at least pypi, homebrew, pacman, chocolatey, winget, conda.
